docs/hub/how-to-upstream.md

```diff
@@ -0,0 +1,276 @@
+
+## `HfApi`
+
+The `HfApi` class is a high-level class that wraps around HTTP requests. There are many valuable tasks you can accomplish with the `HfApi` class, including: 
+
+- List and filter models and datasets.
+- Inspect model or dataset metadata.
+- Delete a repository.
+- Change the visibility of a repository.
+
+### List and filter
+
+It can be helpful for users to see a list of available models and filter them according to a specific language or framework. This can be especially useful for library and organization owners who want to view all their models. Use the `list_models` function with the `filter` parameter to search for specific models.
+
+You can view all the available filters on the Hugging Face Hub website.
+
+![/docs/assets/hub/hub_filters.png](/docs/assets/hub/hub_filters.png)
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List all models.
+>>> api.list_models()
+
+# List only text classification models.
+>>> api.list_models(filter="text-classification")
+
+# List only Russian models compatible with PyTorch.
+>>> api.list_models(filter=("ru", "pytorch"))
+
+# List only the models trained on the "common_voice" dataset.
+>>> api.list_models(filter="dataset:common_voice")
+
+# List only the models from the spaCy library.
+>>> api.list_models(filter="spacy")
+```
+
+Explore available public datasets with `list_datasets`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# List only text classification datasets.
+>>> api.list_datasets(filter="task_categories:text-classification")
+
+# List only datasets in Russian for language modeling.
+>>> api.list_datasets(filter=("languages:ru", "task_ids:language-modeling"))
+```
+
+### Inspect model or dataset metadata
+
+Get important information about a model or dataset as shown in the following:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+# Get metadata of a single model.
+>>> api.model_info("distilbert-base-uncased")
+
+# Get metadata of a single dataset.
+>>> api.dataset_info("glue")
+```
+
+### Create a repository
+
+Create a repository with `create_repo` and give it a name with the `name` parameter:
+
+1. Get your Hugging Face API token:
+
+   ```python
+   >>> from huggingface_hub import HfFolder
+   >>> token = HfFolder().get_token()
+   ```
+2. Use `create_repo` to create your repository:
+
+   ```python
+   >>> from huggingface_hub import HfApi
+   >>> api = HfApi()
+   >>> api.create_repo(token=token, name="test-model")
+   'https://huggingface.co/lysandre/test-model'
+   ```
+
+### Delete a repository
+
+Delete a repository with `delete_repo`. Make sure you are certain you want to delete a repository because this is an irreversible process!
+
+Pass your token and the name of the repository to `delete_repo`:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.delete_repo(token=token, name=REPO_NAME)
```

Comment:
Would it make sense to explain that `REPO_NAME` is the full repository ID of `{username_or_org}/{repo_name}` and that you can get the full string using `HfAPI().get_full_repo_name(repo_name)`?

Reply:
Yes, I thought the `REPO_NAME` was just the repository name. Didn't realize it also includes the username or organization. Thanks for clarifying!